### PR TITLE
Restores buckling sanity.

### LINKED ISSUE #32287: Spooky ghosts can pull people onto vehicles
Issue reported from Round ID: 78175 (/tg/Station Bagil [ENGLISH] [100% FREE LAG])

See title,
Discovered that ghosts can click-drag people onto things like the janicart for loads of SPOOKY FUN.
(Probably works for other vehicles too!)

2SPOOKY/FUN things you can do: 
Kidnap people from maint into an autism fort, drag the HoP from behind his desk, keep the janitor from leaving the custodial closet, and certainly more.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #32287

### CODE REVIEW

File: code/modules/vehicles/vehicle.dm [hunk 1]
```diff
@@ -35,8 +35,8 @@
 		. = ..()
 
 
-/obj/vehicle/user_buckle_mob(mob/living/M, mob/user)
-	if(user.incapacitated())
+/obj/vehicle/user_buckle_mob(mob/living/M, mob/living/user)
+	if(!istype(user) || user.incapacitated())
```
Comment (JJRcop) [inline code/modules/vehicles/vehicle.dm]: If this sanity is called by another proc that also has sanity, is this sanity insanity?

Comment (AnturK) [reply code/modules/vehicles/vehicle.dm]: No because buckling does not need to originate in click code.

Comment (JJRcop) [reply code/modules/vehicles/vehicle.dm]: So then it is an insanity superposition!

Comment (lzimann) [reply code/modules/vehicles/vehicle.dm]: wouldn't it be better to have the sanity check only here then?

Comment (AnturK) [reply code/modules/vehicles/vehicle.dm]: If we had responsible developers yes, but i'll settle for redundancy here for peace of mind.

**Pull request merged**